src/gui/screen_menu_filament.cpp

```diff
@@ -14,7 +14,12 @@
 
 /// Sets temperature of nozzle not to ooze before print (MBL)
 void setPreheatTemp() {
-    marlin_gcode_printf("M104 S%d D%d", (int)PREHEAT_TEMP, (int)filaments[get_filament()].nozzle);
+    const marlin_vars_t *vars = marlin_vars();
+    if (!vars)
+        marlin_gcode_printf("M104 S0");
+
+    /// don't read from EEPROM since it's not in sync
```

Comment:
EEPROM?

Reply:
Previous implementation that does not work.